enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,381 @@
+
+Another risk is that this introduces more deployment configurations which might
+have unforeseen consequences (e.g. not installing the imageregistry causes some
+other component that assumes there is always an imageregistry or assumes the
+presence of some CRD api that is installed with the imageregistry to break).
+
+There was some discussion about the pros/cons of allowing each component to be enabled/disabled independent
+of that component explicitly opting into a particular (presumably well tested) configuration/topology
+[here](https://github.com/openshift/enhancements/pull/200#discussion_r375837903).  The position of this EP is that
+we should only recommend the exclusion of fully independent "capability" components that are not depended on by
+other components.  Further the assumption is that it will be reasonable to tell a customer who disabled
+something and ended up with a non-functional cluster that their chosen exclusions are simply not supported
+currently, or that they must accept the degraded functionality caused by the missing dependency, if they
+intend to keep it disabled.
+
+Since the only components/resources that can be filtered out of the installation are ones that are explicilty
+annotated with `capability.openshift.io/$component`, end-users will not be able to use this mechanism to filter
+components/resources that we did not intend for them to be able to filter out.
+
+## Design Details
+
+### Answered Questions
+
+
+1. Do we want to constrain this functionality to turning off individual capabilities?  We could
+also use it to
+  a) turn on/off groups of components as defined by "solutions" (e.g. a "headless" solution
+  which might turn off the console but also some other capabilities).  This is what CLUSTER_PROFILES
+  sort of enable, but there seems to be reluctance to expand the cluster profile use case to include
+  these sorts of things.
+  b) enable/disable specific configurations/topologies such as "HA", where components could contribute multiple
+  deployment definitions for different configurations and then the installer/CVO would select the correct
+  one based on the chosen install configuration (HA vs single node) instead of having components read/reconcile
+  the infrastructure resource.
+
+Current plan is to constrain this functionality to capability level controls.
+
+2. How does the admin enable a component post-install if they change their mind about what components
+they want enabled?  Do we need/want to allow this?
+
+Turning on a component later is relatively easy (we expose a config resource for the CVO that defines
+the filter, we allow the user to remove items from the filter, the CVO will apply the previously
+filtered resources during the next reconciliation).
+
+Turning off a component later is more problematic because
+a) The CVO doesn't delete resources today, so that would be a new thing to teach it to do.
```

Comment:
Since #419 was implemented, we do.  So this would be "turning off `$COMPONENT` enables some component-deletion manifests", which is a smaller CVO-side lift.